cmd/provisioner-localpv/app/config.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2019 The OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+package app
+
+import (
+	//"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	mconfig "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	cast "github.com/openebs/maya/pkg/castemplate/v1alpha1"
+	"github.com/openebs/maya/pkg/util"
+	//"github.com/pkg/errors"
+	errors "github.com/openebs/maya/pkg/errors/v1alpha1"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//storagev1 "k8s.io/api/storage/v1"
+)
+
+const (
+	//KeyPVStorageType defines if the PV should be backed
+	// a hostpath ( sub directory or a storage device)
+	KeyPVStorageType = "StorageType"
+	//KeyPVBasePath defines base directory for hostpath volumes
+	// can be configured via the StorageClass annotations.
+	KeyPVBasePath = "BasePath"
+	//KeyPVRelativePath defines the alternate folder name under the BasePath
+	// By default, the pv name will be used as the folder name.
+	// KeyPVBasePath can be useful for providing the same underlying folder
+	// name for all replicas in a Statefulset.
+	// Will be a property of the PVC annotations.
+	KeyPVRelativePath = "RelativePath"
+	//KeyPVAbsolutePath specifies a complete hostpath instead of
+	// auto-generating using BasePath and RelativePath. This option
+	// is specified with PVC and is useful for granting shared access
+	// to underlying hostpaths across multiple pods.
+	KeyPVAbsolutePath = "AbsolutePath"
+)
+
+const (
+	// Some of the PVCs launched with older helm charts, still
+	// refer to the StorageClass via beta annotations.
+	betaStorageClassAnnotation = "volume.beta.kubernetes.io/storage-class"
+)
+
+//getVolumeConfig creates a new VolumeConfig struct by
+// parsing and merging the configuration provided in the PVC
+// annotation - cas.openebs.io/config with the
+// default configuration of the provisioner.
+func (p *Provisioner) GetVolumeConfig(pvName string, pvc *v1.PersistentVolumeClaim) (*VolumeConfig, error) {
+
+	pvConfig := p.defaultConfig
+
+	//Fetch the SC
+	scName := GetStorageClassName(pvc)
+	sc, err := p.kubeClient.StorageV1().StorageClasses().Get(*scName, metav1.GetOptions{})
```

Comment:
A wrapper for storageclass exists -- you can get rid of clientsets from here. ( Can take this later also.)